### PR TITLE
Error message updated in case of directory create failure when parent directory doesn't exists or having only read-only permission.

### PR DESCRIPTION
**Pull request by ashwin-msys**
Signed-off-by: Ashwin <avarma@msystechnologies.com>

## Description

Hi.. 

With all my understanding I checked all these scenarios where I was able to reproduce the issue.

Here are the following scenarios. (Screenshots attached below.)

**Case 1:**
 - When the parent directory exists.
   - If the parent directory exists and is writable as well then we won't face any issue. Everything will look good. Please refer the 
     Screenshot below. 
 - Screenshot:
   Screenshot from 2021-07-12 15-49-54

**Case 2:** 
 - When the parent directory exists but having on "read only" permission.
   - In this case we will be facing insufficient permission issue which is correct.
 - Screenshot:
   Screenshot from 2021-07-12 16-26-48

**Case 3:** 
 - When the parent directory doesn't exists.
   - In this case also we are seeing the insufficient permission issue but I guess we should see the different error message like - "Parent Directory does not exists"
   - I have made these changes accordingly. Please refer the Screenshot
 - Screenshots:
  Screenshot from 2021-07-12 15-46-27
  
**PS: Earlier in case if the directory was does not exists, then also we were seeing the `insufficient permissions error`. But I have added one extra condition to show the `parent directory does not exists error`.**

## Related Issue
#11641 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by lamont-granquist** [comment]
chefstyle errors need fixing

File: lib/chef/provider/directory.rb [hunk 1]
```diff
@@ -102,8 +102,13 @@ def define_resource_requirements
               end
             end
           end
-          a.failure_message(Chef::Exceptions::InsufficientPermissions,
-            "Cannot create #{new_resource} at #{new_resource.path} due to insufficient permissions")
+          if !::File.exist?(parent_directory)
+            a.failure_message(Chef::Exceptions::EnclosingDirectoryDoesNotExist,
+              "Parent directory #{parent_directory} does not exists.")
```
Comment (tas50) [inline lib/chef/provider/directory.rb:107]:
```suggestion
              "Parent directory #{parent_directory} does not exist.")
```

**Comment by tas50** [comment]
@ashwin-msys can you look at your unit test failures here?

**Comment by johnmccrae** [comment]
@johnmccrae note to self - this might not rebase, in which case, close it. If it does rebase, see if it can be merged

**Comment by jaymzh** [comment]
@johnmccrae - ping?

**Pull request closed without merging**